plugin/pkg/admission/resourcequota/config.go

```diff
@@ -37,19 +37,38 @@ func init() {
 	install.Install(scheme)
 }
 
+// AddCriticalPodLimitedResources limits the number of critical pods that can be created. We're creating a default
+// resource quota in `kube-system` namespace to allow unlimited number of critical pods to be created in that
+// namespace. Making this function public for easy testing
+func AddCriticalPodLimitedResources() []resourcequotaapi.LimitedResource {
+	return []resourcequotaapi.LimitedResource{
+		{
+			Resource: "pods",
+			MatchScopes: []corev1.ScopedResourceSelectorRequirement{
+				{
+					ScopeName: corev1.ResourceQuotaScopePriorityClass,
+					Operator:  corev1.ScopeSelectorOpIn,
+					Values:    []string{"system-cluster-critical", "system-node-critical"},
+				},
+			},
+		},
+	}
+}
+
 // LoadConfiguration loads the provided configuration.
 func LoadConfiguration(config io.Reader) (*resourcequotaapi.Configuration, error) {
-	// if no config is provided, return a default configuration
+	// if no config is provided, return a default configuration with critical pods as limited resources
 	if config == nil {
-		externalConfig := &resourcequotav1beta1.Configuration{}
-		scheme.Default(externalConfig)
-		internalConfig := &resourcequotaapi.Configuration{}
-		if err := scheme.Convert(externalConfig, internalConfig, nil); err != nil {
-			return nil, err
-		}
-		return internalConfig, nil
+		config := &resourcequotaapi.Configuration{}
+		scheme.Default(config)
+		// append pods as limited resources so that we can limit the number of critical pods to be created. We have
+		// a matching default quota in `kube-system` namespace which allows unlimited pods to be created in that
+		// namespace
+		config.LimitedResources = append(config.LimitedResources, AddCriticalPodLimitedResources()...)
```

Comment:
What if there is an existing and conflicting `LimitedResource` config? What would be the behavior of ResourceQuota?

Reply:
It may not happen here since we're creating config object here but it can certainly happen at https://github.com/kubernetes/kubernetes/pull/76310/files#diff-468653fefaaca80a88656dcb46002defR87. I thought about this but don't have a clean solution atm because of combinations possible.

Reply:
Yeah, those conflicts could cause surprises. For that reason, I am not comfortable with this change at the moment. I think more checks are needed and if conflicting `LimitedResources` exist, we should either merge them, or at least not add the new rules and throw errors.

Reply:
> if conflicting LimitedResources exist, we should either merge them, or at least not add the new rules and throw errors.

So to be clear, even if we have conflicting `LimitedResources` it won't cause an error, as long as we have a matching limitedresource we would proceed allow `system-critical-pods` to be created in `kube-system` namespace. The only thing that concerns me is should we allow that to happen or not(the combinations I was talking about earlier) and it might not be transparent to cluster-admin as to why this is happening but this has been age old problem and clear documentation on why this happens might solve the problem.

I updated the integration test to have conflicting `LimitedResource` operators(In, NotIn) and still the test passes - https://github.com/kubernetes/kubernetes/pull/76310/files#diff-4b71ba6f24c40153543bb61e06d2e54cR403

Reply:
i am not sure i see an immediate conflict, by definition, a pod with system-critical or system-node critical priority was a limited resource in that it could only be created in one location with an implied quota.  this is basically codifying that rule in the configuration.

Reply:
I agree with you. No one wants to create or would have created a config which tells I want my `critical-pods` to be created in any namespace.